### PR TITLE
Add logic for pilot to read the istio ConfigMap

### LINKED ISSUE #3826: connectTimeout in the configMap does not take effect
I tried to change the connectTimeout from the default 1000ms to a larger value by changing the configMap value, but this has no effect.

#kubectl get configmap istio -n istio-system -oyaml | grep connect
    # Edit this list to avoid using mTLS to connect to these services.
      # TCP connection timeout between Envoy & the application, and between Envoys.
      connectTimeout: 10s

This is what Pilot generates in CDS, connectTimeout is still 1000ms
clusters": [
   {
    "name": "in.8079",
    "connect_timeout_ms": 1000,
    "type": "static",
    "lb_type": "round_robin",
    "hosts": [
     {
      "url": "tcp://127.0.0.1:8079"
     }
    ],
    "features": "http2"
   },
   {
    "name": "in.8080",
    "connect_timeout_ms": 1000,
    "type": "static",
    "lb_type": "round_robin",
    "hosts": [
     {
      "url": "tcp://127.0.0.1:8080"
     }
    ]
   },

### PR DESCRIPTION
**Pull request by andraxylia**
Fixes #3826

Duplicate of #3916 who got into a bad state after rebasing the branch.

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -282,6 +286,33 @@ func (s *Server) initClusterRegistries(args *PilotArgs) (err error) {
 	return err
 }
 
+// GetMeshConfig fetches the ProxyMesh configuration from Kubernetes ConfigMap.
+func GetMeshConfig(kube kubernetes.Interface, namespace, name string) (*v1.ConfigMap, *meshconfig.MeshConfig, error) {
```
Comment (ldemailly) [inline pilot/pkg/bootstrap/server.go]: cool,so you can also remove
https://github.com/istio/istio/blob/master/tests/e2e/tests/pilot/util/environment.go#L98
and
https://github.com/istio/istio/blob/master/tests/e2e/tests/pilot/util/environment.go#L149
?

**Comment by kyessenov** [comment]
Are you no longer enforcing no-use of client-go in agent since it bloats
the binary size 3 times?

File: tests/e2e/tests/pilot/util/environment.go [hunk 2]
```diff
@@ -145,27 +146,6 @@ func NewEnvironment(config Config) *Environment {
...
-// GetMeshConfig fetches the ProxyMesh configuration from Kubernetes ConfigMap.
-func GetMeshConfig(kube kubernetes.Interface, namespace, name string) (*v1.ConfigMap, *meshconfig.MeshConfig, error) { // nolint: lll
-	config, err := kube.CoreV1().ConfigMaps(namespace).Get(name, meta_v1.GetOptions{})
-	if err != nil {
-		return nil, nil, err
-	}
-
-	// values in the data are strings, while proto might use a different data type.
-	// therefore, we have to get a value by a key
-	cfgYaml, exists := config.Data[ConfigMapKey]
```
Comment (ldemailly) [inline tests/e2e/tests/pilot/util/environment.go]: the constant too

**Comment by andraxylia** [comment]
@kyessenov this is just server side, it does not add anything to the agent, does it?

File: tests/e2e/tests/pilot/util/environment.go (hunk 2 above)
Comment (andraxylia) [reply tests/e2e/tests/pilot/util/environment.go]: done, thanks for noticing.

**Comment by ldemailly** [comment]
```
I0310 18:22:23.969] --- FAIL: TestPilot (8.10s)
I0310 18:22:23.969] 	pilot_test.go:89: configmaps "istio" not found
I0310 18:22:23.969] FAIL
I0310 18:22:23.970] FAIL	istio.io/istio/tests/e2e/tests/pilot	8.335s
I0310 18:22:23.991] tests/istio.mk:89: recipe for target 'e2e_pilot' failed
```
the error is caused by/first in this PR - no ? can you fix it ?

**Pull request merged**